pdns/dnsparser.hh

```diff
@@ -289,17 +290,33 @@ protected:
 
 struct DNSRecord
 {
-  DNSRecord() : d_type(0), d_class(QClass::IN), d_ttl(0), d_clen(0), d_place(DNSResourceRecord::ANSWER)
+  DNSRecord() :
+    d_type(0), d_class(QClass::IN), d_place(DNSResourceRecord::ANSWER)
   {}
   explicit DNSRecord(const DNSResourceRecord& rr);
+  DNSRecord(const std::string& name, std::shared_ptr<DNSRecordContent> content, const uint16_t type, const uint16_t qclass = QClass::IN, const uint32_t ttl = 86400, const uint16_t clen = 0, const DNSResourceRecord::Place place = DNSResourceRecord::ANSWER) :
+    d_name(DNSName(name)), d_content(std::move(content)), d_type(type), d_class(qclass), d_ttl(ttl), d_clen(clen), d_place(place) {}
+
   DNSName d_name;
   std::shared_ptr<DNSRecordContent> d_content;
   uint16_t d_type;
-  uint16_t d_class;
-  uint32_t d_ttl;
-  uint16_t d_clen;
+  uint16_t d_class{};
```

Comment:
Why this empty initialization? Isn't it the default? Or are you trying to make that explicit? In that case it make sense.

Reply:
It is to be explicit, and then there's no need to initialize them in the constructors. I believe without assignment, members don't get "zero values" and instead just get pre-existing memory garbage.